docs: add migration guides to sidenav in updating to Version 9 section

ajitsinghkaler opened the pull request:
All migration guides did not have a direct link to access them so added them to the side nav in the section updating to version 9 for direct access, it also helps to add right side nav to these migration guides

Fixes #33582

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
No navigation in migration of renderer to renderer 2

Issue Number: :33582


## What is the new behavior?
Navigation available in the migration guide

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

gkalpak reviewed (commented):
What about other migration docs? It might make sense to group them all together.

ajitsinghkaler commented:
@gkalpak how can I find all the migration guides

I agree we can create a tab in the side panel for migration guides.

@kapunahelewong do we have a list of all the migration guides

gkalpak commented:
> @gkalpak how can I find all the migration guides

No, idea :grin:
I am sure @kapunahelewong knows more on that one :wink:

> I agree we can create a tab in the side panel for migration guides.

Just to be clear, I am suggesting keeping them `hidden`, but yeah, it would make sense to have them all in one group in the sidenav.

kapunahelewong commented:
We did discuss having them in the nav or not, with @IgorMinar and @kara recommending that they stay out and remain links from `deprecations.md`. The thought was that as we move forward through versions that deprecations would naturally grow yet become less relevant as time passed, which would in turn necessitate frequent changing of the left nav. Does that help clarify, @ajitsinghkaler?

ajitsinghkaler commented:
Thanks @kapunahelewong once you rename them should we add all of them to the sidenav with hidden true directive this will make the right side nav available for them and left sidenav will also not keep changing continously

kapunahelewong commented:
@ajitsinghkaler, I mean as I created them I named them with `migration-` to begin with (but I think there are a couple that I missed/were there already). I try not to rename files to avoid redirects. Regarding adding them as hidden so they will get the right hand TOC sounds like a great idea.

ajitsinghkaler commented:
@gkalpak @kapunahelewong made the changes please have a look

mary-poppins commented:
You can preview 0f5dc5e at https://pr34979-0f5dc5e.ngbuilds.io/.

gkalpak commented:
@ajitsinghkaler, can you please make CI green?

mary-poppins commented:
You can preview 4c28408 at https://pr34979-4c28408.ngbuilds.io/.

mary-poppins commented:
You can preview 95eefa1 at https://pr34979-95eefa1.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak made the changes please have a look

gkalpak reviewed (approved):
Thx, @ajitsinghkaler! Please change the commit message to reflect the fact that all migration guides are added to the sidenav; e.g.:

> docs: add sidenav and ToC for all migration guides
>
> ...

aikithoughts reviewed (changes requested):
As this content is really part of the updating process, can we move the TOC entries to child elements under the Updating to Version 9 (line 706, I think?)

Also, I don't think these topics should be hidden. If they're important enough to publish, they're important enough to ensure that users can find them. And, when they do find them, they should be able to navigate back to other topics easily (without using the back button. :) )

ajitsinghkaler commented:
@aikidave @kapunahelewong already discussed it with @IgorMinar @kara https://github.com/angular/angular/pull/34979#issuecomment-579379393

they told us we cannot put them In the side nav.

And all of this content is not really updating to version 9 some of them are just in general migration guides for earlier versions too

aikidave commented:
> @aikidave @kapunahelewong already discussed it with @IgorMinar @kara [#34979 (comment)](https://github.com/angular/angular/pull/34979#issuecomment-579379393)
> 
> they told us we cannot put them In the side nav.
> 
> And all of this content is not really updating to version 9 some of them are just in general migration guides for earlier versions too

I'm happy to discuss this with folks, and I'm certainly willing to have my mind changed. But I don't think the solution "temporary" topics is to publish them but not let anyone know that they exist save through a direct link. I don't think it's hard to review content and decide that it's no longer relevant. It happens all the time. :)

I'm new to Angular, so I'll defer to you regarding whether this content is relevant to other updates. But, even so, our docs only talk about updating to version 9, so it still makes sense for this content to exist under that topic.

ajitsinghkaler commented:
> > @aikidave @kapunahelewong already discussed it with @IgorMinar @kara [#34979 (comment)](https://github.com/angular/angular/pull/34979#issuecomment-579379393)
> > they told us we cannot put them In the side nav.
> > And all of this content is not really updating to version 9 some of them are just in general migration guides for earlier versions too
> 
> I'm happy to discuss this with folks, and I'm certainly willing to have my mind changed. But I don't think the solution "temporary" topics is to publish them but not let anyone know that they exist save through a direct link. I don't think it's hard to review content and decide that it's no longer relevant. It happens all the time. :)
> 
> I'm new to Angular, so I'll defer to you regarding whether this content is relevant to other updates. But, even so, our docs only talk about updating to version 9, so it still makes sense for this content to exist under that topic.

Can you please talk to them about this I already proposed to put it in the side nav. As we have no links to reach these migration guides.

kapunahelewong commented:
@ajitsinghkaler, yes, that was the context/history of the decision to have them not in the nav, but since then @aikidave has joined as our docs team lead, so I'll have to defer to him. This PR got the docs team really talking about the concept of hidden nav entries, so thank you very much for bringing it up. Going forward, we can follow @aikidave's lead. :)

ajitsinghkaler commented:
thanks for clearing it out I'll make the changes and push it.

aikidave commented:
Thanks, @ajitsinghkaler! I am sorry I didn't think to introduce myself, and I should have! Really appreciate your contributions!

ajitsinghkaler commented:
No problem @aikidave we are all trying to improve angular at the end of the day. 

I've also made the changes you can have a look

mary-poppins commented:
You can preview d003181 at https://pr34979-d003181.ngbuilds.io/.

ajitsinghkaler commented:
@aikidave can you please review this

aikidave commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,45 @@
+          "children": [
+            {
+              "url": "guide/updating-to-version-9",
+              "title": "General Version 9 update Guide",
```
Let's call this "Overview" instead of "General Version 9 update Guide"

kara replied:
+1

aikidave commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,45 @@
+              "tooltip": "Everything you need to know for updating your application from version 8 to 9."
+            },
+            {
+              "url": "guide/migration-renderer",
```
I would put all of these under one heading:

"Migration details"

Then, for each child element, you can shorten the TOC title to remove "migration". So

Renderer to Renderer2 migration

becomes

Renderer to Renderer2

Note: We shouldn't change the topic titles--this is just in the TOC.

This change should help make sure folks know that these details aren't necessary for every migration.

kara replied:
Yeah, totally agree

kara reviewed (changes requested):
@aikidave Can we discuss the nav more offline? I think it's a bit odd to give migration guides top billing, but ignore other guides that are more important like the Ivy compatibility guide. These migration guides are more like FYIs, so it makes more sense to me to keep the main "updating to version 9" page as the central page that links out to others.

kara commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,45 @@
+              "url": "guide/updating-to-version-9",
+              "title": "General Version 9 update Guide",
+              "tooltip": "Everything you need to know for updating your application from version 8 to 9."
+            },
```
Can we also add a link to the Ivy compatibility guide on top of the migration folder?

kara reviewed (commented):
@aikidave and I synced on the reasons behind the nav, so I'm also good on having it once the below changes are made.

ajitsinghkaler commented:
@aikidave saw you started working on this. Should I still resolve all the comments by you or not?

Should I continue to work on this or you will be working on this?

aikidave commented:
@ajitsinghkaler I'm glad you saw that I started working on this! It got a little crazy. I was working with @kara about how to best add the migration guides to the table of contents, and things sort of escalated. 

We think the right thing to do is to add the guides as you can see in #35145. But, to do that, we also have to update the SCSS so we can support 4 levels in the table of contents.

Would you like to continue working on this? If so, you'd just need to replicate what I've been doing in #35145. I'm also happy to continue working on this if you'd rather tackle something else. Let me know--I'm good either way, although I admit I'm not sure if folks will be okay with my SCSS changes. :)

ajitsinghkaler commented:
@aikidave I can complete this no problem

aikidave commented:
Sounds good. Do you want me to push my changes to your PR? I think @kapunahelewong showed me how to do that.

There's still some debate on the best way to present this information to users. I'll see if I can get that sorted out today.

ajitsinghkaler commented:
okay push your changes I'll squash our commits

aikidave commented:
@ajitsinghkaler , I've pushed my changes to your branch. Please make sure to review my PR (#35145)  in case there are any additional comments that we should address.

ajitsinghkaler commented:
okay @aikidave update me if anything finalizes regarding this

mary-poppins commented:
You can preview ffb3b45 at https://pr34979-ffb3b45.ngbuilds.io/.

ajitsinghkaler commented:
@akidave I propose new headings in the navigation

Optional v9 migration details-> Optional v9 migration
Missing @Injectable() Decorators->@Injectable() Decorators or @Injectable it as apparent that it is a decorator as it used decorator syntax
$localize Global Import->$localize
Missing ModuleWithProviders Generic-> ModuleWithProviders
@Directive()/@Component() Decorators-> @Directive()/@Component()

This will make our headings quite small without compromising on the information given out by these headings

ajitsinghkaler commented:
@aikidave @kara Can you please review this again

aikidave commented:
@ajitsinghkaler , it looks like you might need to rebase. That should take care of the preview issue.

mary-poppins commented:
You can preview d39766a at https://pr34979-d39766a.ngbuilds.io/.

ajitsinghkaler commented:
@aikidave the preview is generated can you look at the pr now

aikithoughts reviewed (approved):
Looks fine to me. Should have @kara take a look as well to be sure.

ajitsinghkaler commented:
@kara can you please review this

ajitsinghkaler commented:
@kara @aikidave its been a week can you please review this

ajitsinghkaler commented:
@kara can you please review this

kara commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,56 @@
+                },
+                {
+                  "url": "guide/migration-undecorated-classes",
+                  "title": "@Directive() /@Component() Decorators",
```
Might simplify to:

```suggestion
                  "title": "Missing @Directive() Decorators",
```

kara commented on aio/angular.json:
```diff
@@ -5,8 +5,7 @@
       "typescriptMismatch": false
-    },
-    "analytics": false
+    }
```
Revert? Unrelated to your change.

kara commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,56 @@
+                {
+                  "url": "guide/migration-dynamic-flag",
+                  "title": "Dynamic Queries Flag",
+                  "tooltip": "Migration to remove unnecessary`static: false` flag from @ViewChild and @ContentChild queries."
```
```suggestion
                  "tooltip": "Migration to remove unnecessary `static: false` flag from @ViewChild and @ContentChild queries."
```

kara reviewed (commented):
A few more tiny things

ajitsinghkaler commented:
@kara made the changes please have a look

mary-poppins commented:
You can preview e44f8c8 at https://pr34979-e44f8c8.ngbuilds.io/.

ajitsinghkaler commented:
@kara can you please review this

ajitsinghkaler commented:
@kara can you please review this

ajitsinghkaler commented:
@kara can you please review this.

ajitsinghkaler commented:
@kara can you please review

ajitsinghkaler commented:
@kara @IgorMinar can you please review this

ajitsinghkaler commented:
@IgorMinar @kara can you please review this

ajitsinghkaler commented:
@IgorMinar @kara can you please review this

ajitsinghkaler commented:
@kara @IgorMinar can you please have a look

kara commented on aio/content/navigation.json:
```diff
@@ -704,9 +704,56 @@
+              "tooltip": "Details to help you make sure your application is compatible with Ivy."
+            },
+            {
+              "title": "Optional v9 migration details",
```
```suggestion
              "title": "Optional v9 Migration Details",
```

kara reviewed (approved):
LGTM once caps are fixed

ajitsinghkaler commented:
@kara made the changes

ajitsinghkaler committed:
docs: add migration guides to sidenav in updating to Version 9

All migration guides did not have a direct link to access them so added them to the side nav in the section updating to version 9 for direct access, it also helps to add right side nav to these migration guides

Fixes #33582

ajitsinghkaler commented:
I just messed up rebasing please ignore it was a rebased a long time ago. Ignore if any review request sent to you

mary-poppins commented:
You can preview 18b3efb at https://pr34979-18b3efb.ngbuilds.io/.

ajitsinghkaler commented:
The final typo was also resolved @kara @aikidave I think we can add the merge label

IgorMinar commented on aio/content/navigation.json:
```diff
@@ -772,9 +772,56 @@
+              "tooltip": "Details to help you make sure your application is compatible with Ivy."
+            },
+            {
+              "title": "Optional v9 Migration Details",
```
Could you please make this shorter so that it doesn't wrap on narrow screens? How about just  "Optional Migrations"?

IgorMinar reviewed (approved):
One small tweak requested but otherwise lgtm.

ajitsinghkaler committed:
fixup! docs: add migration guides to sidenav in updating to Version 9

mary-poppins commented:
You can preview f45d9e9 at https://pr34979-f45d9e9.ngbuilds.io/.

ajitsinghkaler commented:
@IgorMinar you can remove the cleanup label

IgorMinar commented:
Thanks @ajitsinghkaler !
